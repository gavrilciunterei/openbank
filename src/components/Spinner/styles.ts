import styled from 'styled-components';
import colors from '../../styles/colors';

export const StyledSpinner = styled.svg`
  animation: rotate 2s linear infinite;

  width: 50px;
  height: 50px;

  & .path {
    stroke: ${colors.primary_color};
    stroke-linecap: round;
    animation: dash 1.5s ease-in-out infinite;
  }

  @keyframes rotate {
    100% {
      transform: rotate(360deg);
    }
  }
  @keyframes dash {
    0% {
      stroke-dasharray: 1, 150;
      stroke-dashoffset: 0;
    }
    50% {
      stroke-dasharray: 90, 150;
      stroke-dashoffset: -35;
    }
    100% {
      stroke-dasharray: 90, 150;
      stroke-dashoffset: -124;
    }
  }
`;

export const SpinnerContainer = styled.div`
  display: flex;
  justify-content: center;
`;
