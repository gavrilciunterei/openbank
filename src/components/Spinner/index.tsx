import { SpinnerContainer, StyledSpinner } from './styles';
function Spinner() {
  return (
    <SpinnerContainer>
      <StyledSpinner viewBox="0 0 50 50">
        <circle
          className="path"
          cx="25"
          cy="25"
          r="20"
          fill="none"
          strokeWidth="4"
        />
      </StyledSpinner>
    </SpinnerContainer>
  );
}

export default Spinner;
