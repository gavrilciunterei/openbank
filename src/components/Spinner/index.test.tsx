import React from 'react';
import renderer from 'react-test-renderer';
import Spinner from './index';

it('Test Spinner', () => {
  const tree = renderer.create(<Spinner />).toJSON();
  expect(tree).toMatchSnapshot();
});
