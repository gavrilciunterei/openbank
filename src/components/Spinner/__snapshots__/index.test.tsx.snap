// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Spinner 1`] = `
<div
  className="sc-gsDKAQ eVaFoO"
>
  <svg
    className="sc-bdvvtL fSlqhq"
    viewBox="0 0 50 50"
  >
    <circle
      className="path"
      cx="25"
      cy="25"
      fill="none"
      r="20"
      strokeWidth="4"
    />
  </svg>
</div>
`;
